imap_processing/lo/l0/science_direct_events.py

```diff
@@ -76,45 +65,54 @@ class ScienceDirectEvents(LoBase):
 
     SHCOARSE: int
     COUNT: int
-    DATA: bitstring.Bits
+    DATA: str
     CHKSUM: int
-    TOF0: float
-    TOF1: float
-    TOF2: float
-    TOF3: float
-    TIME: float
-    ENERGY: float
-    POS: float
+    TOF0: np.ndarray
+    TOF1: np.ndarray
+    TOF2: np.ndarray
+    TOF3: np.ndarray
+    TIME: np.ndarray
+    ENERGY: np.ndarray
+    POS: np.ndarray
 
     def __init__(self, packet, software_version: str, packet_file_name: str):
         """Intialization method for Science Direct Events Data class."""
         super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
         self.parse_data(packet)
+        self.TOF0 = np.array([])
+        self.TOF1 = np.array([])
+        self.TOF2 = np.array([])
+        self.TOF3 = np.array([])
+        self.TIME = np.array([])
+        self.ENERGY = np.array([])
+        self.POS = np.array([])
         self._decompress_data()
 
     def _decompress_data(self):
         """Decompress the Lo Science Direct Events data."""
-        case_number = self._find_decompression_case()
-        tof_decoder = self._find_tof_decoder_for_case(case_number)
-        tof_calculation_binary = self._read_tof_calculation_table(case_number)
-        remaining_bit_coefficients = self._find_remaining_bit_coefficients(
-            tof_calculation_binary
-        )
-        parsed_bits = self._parse_binary(case_number, tof_decoder)
-        # decode_fields will set the TOF class variables
-        self._decode_fields(remaining_bit_coefficients, parsed_bits)
+        bitstream = bitstring.ConstBitStream(bin=self.DATA)
```

Comment:
I liked avoiding the bitstream manipulation you had before if possible. I think this is so you can read x bits off the field and use bitstream to do the indexing for you? I wonder if you could pass in the start/range to your function instead and keep what you had. See below for an example of what I mean.

Reply:
Yeah, I can remove bitstring. It will just make things a little more complicated because different types of DEs require you to check certain bits where other types don't, plus looping over multiple DEs in the packet will require more bookkeeping